java-hello/README.md

```diff
@@ -0,0 +1,280 @@
+# Java Hello on Unikraft
+
+Build and run the Java Hello with the ELF Loader. The ELF Loader uses the Unikraft binary-compatibility layer in order to run native Linux binaries (ELFs) with Unikraft. The ELF has to be PIE (Position-Independent Executable), either static or dynamic.
+The ELF has to be PIE (*Position-Independent Executable*), either static or dynamic.
+
+Follow the instructions below to set up, configure, build and run the ELF Loader.
+Make sure you installed the [requirements](../README.md#requirements).
+
+The Unikraft ELF Loader currently runs on x86_64 (ARM64 work is underway) and on KVM (QEMU and Firecracker) (Xen work is underway).
+
+## Quick Setup (aka TLDR)
+
+For a quick setup, run the commands below.
+Note that you still need to install the [requirements](../README.md#requirements).
+Before everything, make sure you run the [top-level `setup.sh` script](../setup.sh).
+
+To build and run a Hello Linux ELF using the Unikraft ELF Loader, use the commands below:
+
+- Extract the rootfs.tar.gz
+
+```console
+tar -xzvf rootfs.tar.gz
+```
+
+```console
+./setup.sh
```

Comment:
Move the tar command in the block below. There is no point in having 2 command blocks.